### ISSUE TITLE
Rules for image files

### ISSUE THREAD

**Issue by noamross** [petebaker/r-makefile-definitions#1]
Thank you for this set of Make rules, which I learned about from your User2016 talk.

I suggest including a set of rules for stand-alone image files.  Often for a project I have scripts to produce just image files (for instance, figures rendered at sizes and resolutions required for journal publication).  A simple set of rules to say `Figure01.svg` and `Figure02.png` are generated by running `Figures.R` would be neat.  If I really need this I'll submit a PR later this week.

**Comment by petebaker** [comment]
Hi Noam
I'm at IBC2016 conference in Victoria BC this week and back to Oz next week. Will look at this then.
As mentioned at UseR, it is an interesting prob with R able to produce many different output files.
My only concern about extra standard rules to make .png, .svg etc generally rather than one-offs because these can also be dependencies for .Rmd files etc. It's easy to get into loops for instance if not careful but undoubtedly fine if well documented. I tend to just set a rule for .Rout file and produce other files as a side effect of running R. The resulting .png, .svg files can still of course be dependencies for other output - eg report.html from report.Rmd.

``` makefile
.PHONY: all
all: report.html
report.html: report.Rmd figs.Rout
figs.Rout: figs.R data.file other.stuff
include ~/lib/common.mk
```

Then at prompt use

``` bash
make
```

Of course that's not so explicit but you won't need to run R twice to get graphics files. Happy to get your thoughts on this
Cheers
Peter

**Comment by petebaker** [comment]
Hi Noam

This must be one of the longest ever issues to close to close on GitHub. I really should have put this up several years ago but overlooked it.

I certainly appreciated your comments here and at UseR and addressed it in the JSS code-snippet published in August. Unfortunately, it seems to have taken a lifetime to get published despite being accepted ages ago. Inspired by your questions, I included Section 5.6 on multiple targets, including graphics files. See  [https://doi.org/10.18637/jss.v094.c01](https://doi.org/10.18637/jss.v094.c01)

Thanks again

Cheers
Peter
